plugin/src/main/java/org/opensearch/ml/action/mcpserver/OpenSearchMcpServerTransportProvider.java

```diff
@@ -113,35 +125,106 @@ public Mono<HttpChunk> handleSseConnection(StreamingRestChannel channel, boolean
             OpenSearchMcpSessionTransport sessionTransport = new OpenSearchMcpSessionTransport(channel);
             McpServerSession session = sessionFactory.create(sessionTransport);
             String sessionId = session.getId();
-            ActionListener<IndexResponse> actionListener = ActionListener.wrap(r -> {
-                if (r != null && r.status() == RestStatus.CREATED) {
-                    log.debug("Created new SSE connection for session: {}", sessionId);
-                    sessions.put(sessionId, session);
-
-                    // Send initial endpoint event
-                    log.debug("Sending initial endpoint event to session: {}", sessionId);
-                    String result;
-                    if (appendToBaseUrl) {
-                        result = String.format(Locale.ROOT, "/_plugins/_ml/mcp/sse/message?sessionId=%s", sessionId);
-                    } else {
-                        result = String.format(Locale.ROOT, "/sse/message?sessionId=%s", sessionId);
-                    }
-                    McpAsyncServerHolder.CHANNELS.put(sessionId, channel);
-                    sink.success(createHttpChunk(ENDPOINT_EVENT_TYPE, result));
+            ActionListener<Boolean> initIndexListener = ActionListener.wrap(created -> {
+                if (created) {
+                    log.debug("Successfully created MCP session management index");
+                    addSession(sessionId, session, appendToBaseUrl, nodeId, client, channel, sink);
                 } else {
-                    log.error("Failed to create new SSE connection for session: {}", sessionId);
-                    sink.error(new IllegalStateException("Failed to create new SSE connection for session" + sessionId));
+                    log.debug("Failed to create MCP session management index for session: {}", sessionId);
+                    sink
+                        .error(
+                            new IllegalStateException(
```

Comment:
Why is this a 4XX level error?

Reply:
I think `IllegalStateException` is a 5XX level error.